spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteMergeInto.scala

```diff
@@ -58,53 +61,136 @@ case class RewriteMergeInto(conf: SQLConf) extends Rule[LogicalPlan] with Rewrit
 
   override def apply(plan: LogicalPlan): LogicalPlan = {
     plan resolveOperators {
+      case MergeIntoTable(target: DataSourceV2Relation, source: LogicalPlan, cond, matchedActions, notMatchedActions)
+          if matchedActions.isEmpty =>
+
+        val (_, targetTableScan) = buildTargetScan(target, source, cond)
+
+        // when there are no matched actions, use a left anti join to remove any matching rows and rewrite to use
+        // append instead of replace. only unmatched source rows are passed to the merge and actions are all inserts.
+        val joinPlan = Join(source, targetTableScan, LeftAnti, Some(cond), JoinHint.NONE)
+
+        val mergeParams = MergeIntoParams(
+          isSourceRowNotPresent = FALSE_LITERAL,
+          isTargetRowNotPresent = TRUE_LITERAL,
```

Comment:
As a follow-up, we should probably negate these conditions because it is harder to reason about "not present".

Reply:
I agree on negating these conditions, possibly by having a shim we use for our own use cases. Having just reasoned about "not present" in `core` for NOT_STARTS_WITH, I can confirm that it is a pain and leads to confusion.

Reply:
+1

Reply:
Resolving this as we won't address in this PR.